### PR TITLE
Input select

### PR DESCRIPTION
**Pull request by allan-chagas-brisa**
## Description

This PR implements the new component Input Select, which consists in a input next to a select button with a label that describes its usage.

## Proposed Changes

- User can pass the properties to the component via input following the interface IonInputSelectProps which has the following properties: 
@Input() name: string; The property to link the input element with an possible label;
@Input() disabled = false; The property to disable the button and the input;
@Input() value = ''; The property that holds the value of the first input;
@Input() secondValue = ''; The property that holds the value of the second input;
@Input() selectOptions: SelectOption[] = defaultSelectOptions; The property where the user passes an array of options, of the type SelectOption, to be displayed on the dropdown from the select component;
@Output() valueChange = new EventEmitter<ValueToEmmit>(); The emmiter, that emmits an object of the type ValueToEmmit, every time the user inputs in the element;

## How to Test

yarn test input-select

## Screenshots

[Gravação de tela de 06-10-2023 10:26:36.webm](https://github.com/Brisanet/ion/assets/138060158/e0604f2b-3eec-404d-97a3-e559c72822ef)

## View Storybook

[Storybook](https://62eab350a45bdb0a5818520e-lvpnlhmvcf.chromatic.com/?path=/story/ion-data-entry-input-select--input-select)

## Compliance

- [X] I have verified that this change complies with our code and contribution policies.
- [X] I have verified that this change does not cause regressions and does not affect other parts of the code.

### CODE REVIEW

File: projects/ion/src/lib/input-select/input-select.component.html [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        <input
+          data-testid="second-input"
+          [id]="name"
+          type="text"
+          class="my-input"
+          [attr.placeholder]="
+            currentOption?.secondPlaceholder
+              ? currentOption.secondPlaceholder
+              : 'Valor Final'
+          "
```
Comment (iurynogueira) [inline projects/ion/src/lib/input-select/input-select.component.html:58]:
```suggestion
          [attr.placeholder]="
            currentOption?.secondPlaceholder || 'Valor Final'
          "
```

File: projects/ion/src/lib/input-select/input-select.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+const resetComponentState = (): void => {
+  const selectButton = screen.getByTestId('ion-select-button');
+  fireEvent.click(selectButton);
+  const firstOption = document.getElementById('option-0');
+  if (firstOption) {
+    fireEvent.click(firstOption);
+  }
+};
+
+const selectOptionWithmultple = (): HTMLElement => {
```
Comment (iurynogueira) [inline projects/ion/src/lib/input-select/input-select.component.spec.ts:27]:
```suggestion
const selectOptionWithMultiple = (): HTMLElement => {
```

**Review by iurynogueira — APPROVED** [review]

File: projects/ion/src/lib/input-select/input-select.component.ts [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+  @Input() disabled = false;
+  @Input() value = '';
+  @Input() secondValue = '';
+  @Input() selectOptions: SelectOption[] = defaultSelectOptions;
+  @Output() valueChange = new EventEmitter<ValueToEmmit>();
+
+  public dropdownVisible = false;
+  public currentOption!: SelectOption;
+
+  public handleSelect(selectedOption: SelectOption[]): void {
```
Comment (jairochaves2) [inline projects/ion/src/lib/input-select/input-select.component.ts:44]: When select the option, it keeps the text secondValue.

Maybe he should clear both values when changing the option.

image

File: projects/ion/src/lib/input-select/input-select.component.ts [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+  }
+
+  public handleClick(): void {
+    this.toggleDropdownVisibility();
+  }
+
+  public handleChange(): void {
+    this.valueChange.emit({
+      optionSelected: this.currentOption,
+      inputValue: this.value,
```
Comment (jairochaves2) [inline projects/ion/src/lib/input-select/input-select.component.ts:56]:
```suggestion
      firstValue: this.value,
```

To maintain the placeholder default

File: projects/ion/src/lib/input-select/input-select.component.ts [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+  },
+  {
+    label: 'Maior ou igual a',
+  },
+  {
+    label: 'Menor que',
+  },
+  {
+    label: 'Menor ou igual a',
+  },
```
Comment (jairochaves2) [inline projects/ion/src/lib/input-select/input-select.component.ts:25]: Put in alphabetical order?

**Review by jairochaves2 — APPROVED** [review]

**Pull request merged**